src/indexer.c

```diff
@@ -106,6 +109,7 @@ static void writeCurEntries(RSAddDocumentCtx *aCtx, RedisSearchCtx *ctx) {
     if (invidx) {
       entry->docId = aCtx->doc->docId;
       RS_LOG_ASSERT(entry->docId, "docId should not be 0");
+      IndexerYieldWhileLoading(ctx->redisCtx);
```

Comment:
Do we really want to potentially yield on **every term in the document**? Even if we don't yield, we call `RedisModule_GetContextFlags(ctx)` every time. This can significantly affect the indexing performance. CC @alonre24

Reply:
I agree that it may add some overhead. Looking at the benchmark report, I don't see anything unusual on load flows. But on the other hand, it is not that reliable either... So I guess it does make sense to test it once and avoid jumping to redis code frame each time

Reply:
I don't think we have any benchmark scenario for indexing real-world text dataset, keep me honest